src/userStatsMap.ts

```diff
@@ -0,0 +1,117 @@
+import {
+	ClearingHouse,
+	UserAccount,
+	getUserStatsAccountPublicKey,
+	OrderRecord,
+} from '@drift-labs/sdk';
+import { ProgramAccount } from '@project-serum/anchor';
+
+import { PublicKey } from '@solana/web3.js';
+
+export class UserStatsMap {
+	/**
+	 * map from UserAccount pubkey to ClearingHouseUserStats pubkey
+	 * note: multiple UserAccount may have the same ClearingHouseUserStats since ClearingHouseUserStats are unique
+	 * to the authority, we key by UserAccount to make lookup easier
+	 */
+	private userStatsMap = new Map<string, PublicKey>();
+	private clearingHouse: ClearingHouse;
+
+	constructor(clearingHouse: ClearingHouse) {
+		this.clearingHouse = clearingHouse;
+	}
+
+	public async fetchAllUsers() {
+		const programUserAccounts =
+			(await this.clearingHouse.program.account.user.all()) as ProgramAccount<UserAccount>[];
+
+		for (const programUserAccount of programUserAccounts) {
+			const user: UserAccount = programUserAccount.account;
+
+			if (this.userStatsMap.has(programUserAccount.publicKey.toString())) {
+				continue;
+			}
+
+			this.addUser(programUserAccount.publicKey, user.authority);
+		}
+	}
+
+	private async getUserAccountFromPublicKey(
+		userAccountPublicKey: string
+	): Promise<ProgramAccount<UserAccount>> {
+		const programUserAccounts =
+			(await this.clearingHouse.program.account.user.all()) as ProgramAccount<UserAccount>[];
```

Comment:
you can just call `await this.clearingHouse.program.account.userStats.fetchl({public key})` to fetch a single account

`await this.clearingHouse.program.account.user.all()` is really resource intensive for rpc so should avoid trying to call it when you can